### PR TITLE
Added ability to omit model reference when specifying association inc…

### PR DESCRIPTION
**Pull request by irond13**
…ludes #3193

### CODE REVIEW

File: lib/model.js [hunk 1]
```diff
@@ -289,6 +295,14 @@ function conformInclude(include, self) {
         include.as = include.association.as;
       }
     } else {
+      if (! include.model && include.as) {
+        var association = _.find(self.associations, 'as', include.as);
+        if (! association) throw new Error('Include unexpected. Unable to find Association with \'as\': ' + include.as);
+
+        include.association = association;
+        return conformInclude(include, self);
+      }
```
Comment (mickhansen) [inline lib/model.js]: This would be in the case of `[{as: 'String', where: {}}]`?

**Comment by mickhansen** [comment]
Please add a test that fails when using a string on an association without `as` specified.

File: lib/model.js (hunk 1 above)
Comment (irond13) [inline lib/model.js]: Yes, that's correct (see test 'should allow me refer to an association using only its alias in object form in eager loading')

**Comment by irond13** [comment]
Added two negative test cases

**Comment by janmeier** [comment]
LGTM - I think this is the right approach - if it's in `model.associations`, use it, otherwise fail.

We should remember to document that! :)

**Pull request closed without merging**